### DISCUSSION TITLE
How to inline nullable object types with a reference to not nullable?

### DISCUSSION THREAD

**Discussion by bmaizel** [victools/jsonschema-generator#410] [Q&A] [answered]
Hello,

I have DEFINITIONS_FOR_ALL_OBJECTS set. In the $defs, I get "_definitionName_" or "_definitionName_-nullable" or both, depending on the presence of the `@NotNull` annotation on the particular fields.

What I want, is to always have only "_definitionName_" in the $defs, and nullable field definition inlined as `{"anyOf":[{"type":"null"},{"$ref":"#/$defs/_definitionName_"}]}`

What is the simplest way?

**Comment by CarstenWickner** [comment, marked as answer]
Hi @bmaizel,

So far, there was no simple way.
I've created a new `Option.INLINE_NULLABLE_SCHEMAS` in #412 now in response to your question.

EDIT: after applying the fix in #417, this has been released in [v4.33.0](https://github.com/victools/jsonschema-generator/releases/tag/v4.33.0)

**Comment by bmaizel** [comment]
I just tried it with a snapshot.
If I enable `INLINE_NULLABLE_SCHEMAS`, it very much kills the effect of the `DEFINITIONS_FOR_ALL_OBJECTS`.
Nullable object definitions are completely inlined. What I want, is a nullable reference to a not-nullable object definition.

**Reply by CarstenWickner** [reply]
Good catch. Fixed that now in #417.

**Comment by bmaizel** [comment]
Works as expected in 4.33.0
Thank you, you are great!
